src/main/java/sg/dex/starfish/keeper/DirectPurchaseAdapter.java

```diff
@@ -0,0 +1,111 @@
+package sg.dex.starfish.keeper;
+
+import sg.dex.starfish.impl.squid.SquidService;
+import com.oceanprotocol.common.web3.KeeperService;
+
+import com.oceanprotocol.squid.exceptions.TokenApproveException;
+import java.io.IOException;
+import org.web3j.crypto.CipherException;
+
+import org.web3j.utils.Numeric;
+import java.util.ArrayList;
+import java.math.BigInteger;
+import java.util.Properties;
+import java.io.InputStream;
+import com.oceanprotocol.keeper.contracts.OceanToken;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+
+import org.web3j.protocol.core.methods.request.EthFilter;
+import io.reactivex.Flowable;
+import org.web3j.abi.EventEncoder;
+import org.web3j.protocol.core.DefaultBlockParameter;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+
+public final class DirectPurchaseAdapter {
+    private DirectPurchase directPurchase;
+    private TokenManager tokenManager;
+
+    public DirectPurchaseAdapter() throws IOException, CipherException {
+        // getting properties
+        Properties properties = getProperties();
+        String directPurchaseAddress = (String)properties.getOrDefault("contract.DirectPurchase.address", "");
+        String oceanTokenAddress = (String)properties.getOrDefault("contract.OceanToken.address", "");
+        // getting keeper
+        KeeperService keeper = SquidService.getKeeperService(properties);
+        // loading contract instances
+        directPurchase = DirectPurchase.load(directPurchaseAddress, keeper.getWeb3(), keeper.getTxManager(), keeper.getContractGasProvider());
+        OceanToken oceanToken = OceanToken.load(oceanTokenAddress, keeper.getWeb3(), keeper.getTxManager(), keeper.getContractGasProvider());
```

Comment:
Does this have to be an OceanToken or can any ERC20 Token work?

Reply:
OceanToken object itself has embedded ABI to access OceanToken smart contract. So technically if any else smart contract has the same ABI, the JAVA object OceanToken will be able to call it as well. - this is more low level technical answer.
Regarding the concrete code: Yes, OceanToken Java object is supposed to call only OceanToken smart contract. (it is needed to approve_tokens functionality)